group-coordinator/src/main/java/org/apache/kafka/coordinator/group/RecordHelpers.java

```diff
@@ -467,6 +471,72 @@ public static Record newEmptyGroupMetadataRecord(
         );
     }
 
+    /**
+     * Creates an OffsetCommit record.
+     *
+     * @param groupId           The group id.
+     * @param topic             The topic name.
+     * @param partitionId       The partition id.
+     * @param offsetAndMetadata The offset and metadata.
+     * @param metadataVersion   The metadata version.
+     * @return The record.
+     */
+    public static Record newOffsetCommitRecord(
+        String groupId,
+        String topic,
+        int partitionId,
+        OffsetAndMetadata offsetAndMetadata,
+        MetadataVersion metadataVersion
+    ) {
+        short version = offsetAndMetadata.expireTimestampMs.isPresent() ?
+            (short) 1 : metadataVersion.offsetCommitValueVersion();
+
+        return new Record(
+            new ApiMessageAndVersion(
+                new OffsetCommitKey()
+                    .setGroup(groupId)
+                    .setTopic(topic)
+                    .setPartition(partitionId),
+                (short) 1
```

Comment:
Can we define these `(short) 1` as a constant? That might reduce the changes of us changing one without the others in the future

Reply:
Filed https://issues.apache.org/jira/browse/KAFKA-15225 for this.